connect/transforms/src/test/java/org/apache/kafka/connect/transforms/MaskFieldTest.java

```diff
@@ -250,4 +250,22 @@ public void testReplacementTypeMismatch() {
     public void testEmptyStringReplacementValue() {
         assertThrows(ConfigException.class, () -> checkReplacementSchemaless("short", ""), "String must be non-empty");
     }
+
+    @Test
+    public void testNullListAndMapReplacementsAreMutable() {
+        final List<String> maskFields = new ArrayList<>(SCHEMA.fields().size());
+        SCHEMA.fields().forEach(f -> maskFields.add(f.name()));
+        final Struct updatedValue = (Struct) transform(maskFields, null).apply(record(SCHEMA, VALUES_WITH_SCHEMA)).value();
+        @SuppressWarnings("unchecked") List<Integer> actualList = (List<Integer>) updatedValue.get("array");
+        assertEquals(0, actualList.size());
+        actualList.add(0);
+        List<Integer> expectedList = Collections.singletonList(0);
+        assertEquals(expectedList, actualList);
+
+        @SuppressWarnings("unchecked") Map<String, String> actualMap = (Map<String, String>) updatedValue.get("map");
+        assertEquals(0, actualMap.size());
+        actualMap.put("k", "v");
+        Map<String, String> expectedMap = Collections.singletonMap("k", "v");
+        assertEquals(expectedMap, actualMap);
```

Comment:
Few small things:

- It's unnecessary to mask every field when we're only testing the behavior of the `array` and `map` field, and the test case is clearer if we only mask those two
- Just like with the test for `HoistField`, we should be comparing against empty collections instead of performing assertions on collection size here

```suggestion
        List<String> maskFields = Arrays.asList("array", "map");
        final Struct updatedValue = (Struct) transform(maskFields, null).apply(record(SCHEMA, VALUES_WITH_SCHEMA)).value();

        @SuppressWarnings("unchecked") List<Integer> actualList = (List<Integer>) updatedValue.get("array");
        assertEquals(Collections.emptyList(), actualList);
        actualList.add(0);
        List<Integer> expectedList = Collections.singletonList(0);
        assertEquals(expectedList, actualList);

        @SuppressWarnings("unchecked") Map<String, String> actualMap = (Map<String, String>) updatedValue.get("map");
        assertEquals(Collections.emptyMap(), actualMap);
        actualMap.put("k", "v");
        Map<String, String> expectedMap = Collections.singletonMap("k", "v");
        assertEquals(expectedMap, actualMap);
```

Reply:
Makes sense, done 👍